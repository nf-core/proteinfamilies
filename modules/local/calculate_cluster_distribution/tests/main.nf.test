nextflow_process {

    name "Test Process CALCULATE_CLUSTER_DISTRIBUTION"
    script "../main.nf"
    process "CALCULATE_CLUSTER_DISTRIBUTION"

    test("proteinfamilies - tsv") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'test' ], // meta map
                    file(params.pipelines_testdata_base_path + 'proteinfamilies/test_data/modules/clustering.tsv', checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("proteinfamilies - tsv - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'test' ], // meta map
                    file(params.pipelines_testdata_base_path + 'proteinfamilies/test_data/modules/clustering.tsv', checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
