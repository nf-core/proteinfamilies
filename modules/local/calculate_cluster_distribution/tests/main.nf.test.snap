{
    "proteinfamilies - tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_clustering_distribution_mqc.csv:md5,53d7b52b9ff993372a1153b2ff8ac639"
                    ]
                ],
                "1": [
                    "versions.yml:md5,21017052f4c8e826f6d6ab9fdffe11f5"
                ],
                "mqc": [
                    [
                        {
                            "id": "test"
                        },
                        "test_clustering_distribution_mqc.csv:md5,53d7b52b9ff993372a1153b2ff8ac639"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,21017052f4c8e826f6d6ab9fdffe11f5"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-24T14:43:56.858435601"
    },
    "proteinfamilies - tsv - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_clustering_distribution_mqc.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,21017052f4c8e826f6d6ab9fdffe11f5"
                ],
                "mqc": [
                    [
                        {
                            "id": "test"
                        },
                        "test_clustering_distribution_mqc.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,21017052f4c8e826f6d6ab9fdffe11f5"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-24T14:44:05.560229782"
    }
}